LBBT-Design-Patterns/LBBTCalculator.cs

```diff
@@ -1,60 +1,66 @@
-﻿using System;
-using System.Collections.Concurrent;
-using System.Collections.Generic;
-using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
-
-namespace LBBT_Design_Patterns
+﻿using System.Collections.Concurrent;
+
+namespace LBBT_Design_Patterns;
+
+/// <summary>
+///   Tax calculator using tax bands with rates.
+/// </summary>
+/// <param name="taxStrategies">
+///   Tax strategies to use in calculations,
+///   or will use default strategies specified in <see cref="CreateDefaultStrategies"/> if null.
+/// </param>
+public class LBBTCalculator(IList<ITaxStrategy>? taxStrategies = null) : ITaxCalculator
 {
-    internal class LBBTCalculator
+    private readonly IList<ITaxStrategy> _taxBands = taxStrategies ?? CreateDefaultStrategies();
+    private ConcurrentBag<int> _taxCalculationResults = [];
+    private List<Thread> _threadStore = [];
+
+    private static IList<ITaxStrategy> CreateDefaultStrategies()
     {
-        private List<ITaxStrategy> TaxBands { get; }
-        public int TotalTax { get; private set; } = 0;
-        private List<Thread> threadStore = [];
-        private ConcurrentBag<int> taxCalculationResults = new ConcurrentBag<int>();
-        public LBBTCalculator()
+        TaxBand taxBand2Pc = new(0.02, 145001, 250000);
```

Comment:
naming according to common standards. Local variables lowercase, fields underscore then lowercase, properties uppercase.